modules/fgbio/fastqtobam/main.nf

```diff
@@ -9,29 +9,30 @@ process FGBIO_FASTQTOBAM {
 
     input:
     tuple val(meta), path(reads)
-    val read_structure
 
     output:
-    tuple val(meta), path("*_umi_converted.bam"), emit: umibam
-    path "versions.yml"                         , emit: versions
+    tuple val(meta), path("*.bam"), emit: bam
+    path "versions.yml"           , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def sample_name = args.contains("--sample") ? "" : "--sample ${prefix}"
+    def library_name = args.contains("--library") ? "" : "--library ${prefix}"
+
     """
 
     fgbio \\
         --tmp-dir=. \\
         FastqToBam \\
-        -i $reads \\
-        -o "${prefix}_umi_converted.bam" \\
-        --read-structures $read_structure \\
```

Comment:
Is it fair to remove the `--read-structures` option completely? Are you thinking it should be included via `task.ext.args` when needed?

Reply:
I'd keep it in `ext.args` (like I did in the tests) since it's a non mandatory option, and the tool also supports straight up conversion without UMI parsing